src/main/java/in/projecteka/gateway/registry/RegistryRepository.java

```diff
@@ -0,0 +1,143 @@
+                                monoSink.success(false);
+                                return;
+                            }
+                            monoSink.success(true);
+                        }));
+    }
+
+    public Mono<Void> insertBridgeEntry(BridgeRegistryRequest request) {
+        return Mono.create(monoSink -> dbClient.preparedQuery(INSERT_BRIDGE_ENTRY)
+                .execute(Tuple.of(request.getName(), request.getUrl(), request.getId(),
+                        request.isActive(), request.isBlocklisted()),
+                        handler -> {
+                            if (handler.failed()) {
+                                logger.error(handler.cause().getMessage(), handler.cause());
+                                monoSink.error(new DbOperationError("Failed to insert bridge entry"));
+                                return;
+                            }
+                            monoSink.success();
+                        }));
+    }
+
+    public Mono<Void> updateBridgeEntry(BridgeRegistryRequest request) {
+        return Mono.create(monoSink -> dbClient.preparedQuery(UPDATE_BRIDGE_ENTRY)
+                .execute(Tuple.of(request.getName(), request.getUrl(),
+                        request.isActive(), request.isBlocklisted(), request.getId()),
+                        handler -> {
+                            if (handler.failed()) {
+                                logger.error(handler.cause().getMessage(), handler.cause());
+                                monoSink.error(new DbOperationError("Failed to update bridge entry"));
+                                return;
+                            }
+                            monoSink.success();
+                        }));
+    }
+
+    public Mono<Boolean> ifPresent(String serviceId, ServiceType type, boolean active) {
+        return Mono.create(monoSink -> this.dbClient.preparedQuery(SELECT_ACTIVE_BRIDGE_SERVICE)
+                .execute(Tuple.of(serviceId, type.toString(), active),
+                        handler -> {
+                            if (handler.failed()) {
+                                logger.error(handler.cause().getMessage(), handler.cause());
+                                monoSink.error(new DbOperationError("Failed to fetch active bridge service"));
+                                return;
+                            }
+                            var iterator = handler.result().iterator();
+                            if (!iterator.hasNext()) {
+                                monoSink.success(false);
+                                return;
+                            }
+                            monoSink.success(true);
+                        }));
+    }
+
+    public Mono<Boolean> ifPresent(String serviceId, ServiceType type) {
+        return Mono.create(monoSink -> this.dbClient.preparedQuery(SELECT_BRIDGE_SERVICE)
+                .execute(Tuple.of(serviceId, type.toString()),
```

Comment:
You can reuse right?

Reply:
why do we need two methods?

Reply:
I'll reuse it.. one to check if any active service already exists for that type & other to check whether to insert or update bridge_service record

Reply:
Done